build.js

```diff
@@ -119,8 +176,176 @@ function transpileUsingNestedHandlebars(fileString, fileToCreate, components) {
+		file_content = file_content.replace('{fill_parents}', joinRepeatedString(getNumberOfParentFolders(path,'build') - 1,'../'));
+	}
+	while(file_content.indexOf('{fill_parents_html}') != -1) {
+		file_content = file_content.replace('{fill_parents_html}', joinRepeatedString(getNumberOfParentFolders(path,'html') - 1,'../'));
+	}
+	// console.log("Parent Folders AutoFill Complete On The File: " + path);
+	return file_content;
+}
+
+/** Returns string where language href(url) are replaced with appropriate links
+ * 
+ * @param {String} filepath Path to file where the component is located
+ * @param {*} file_content Content of the file
+ * @returns Returns the file content with transpiled components
+ */
+function languageFillPath(filepath, file_content) {
+	while(file_content.indexOf('{language_src}') != -1) {
+		file_content = file_content.replace('{language_src}', joinPathofFile(filepath));
+	}
+	return file_content;
+}
+
+/** Joins together the same string x number of times
+ * 
+ * @param {Number} multiplier Number how many times the pattern is to be repeated
+ * @param {String} pattern The pattern to be repeatedly joined 
+ * @returns {String} A joined string of the pattern
+ */
+function joinRepeatedString(multiplier, pattern) {
+	var string_builder = "";
+	for(var i = 0; (i + 1) <= multiplier; i++){
+		string_builder += pattern;
+	}
+	return string_builder;
+}
+
+/** Returns the relative url of the file in different language folder
+ * 
+ * @param {String} filepath Path to file where the component is located
+ * @returns {String} The relative url of the same file in different language
+ */
+function joinPathofFile(filepath) {
+	if(getNumberOfParentFolders(filepath) == 0)
+		return;
+	var string_builder = joinRepeatedString(getNumberOfParentFolders(filepath) - 1, '../');
+	var path_parts = filepath.split('html/')[1];
+	if(path_parts.indexOf('en/') != -1)
+		string_builder += path_parts.replace('en/','sk/');
+	else
+		string_builder += path_parts.replace('sk/', 'en/');
+	return string_builder;
+}
+
+/** Get the number of parent folders from the "Root" of HTML files
+ *  
+ * @param {String} path Path of the file 
+ * @param {String} root 
+ * @returns {Number} Number of parent folders from root 
+ */
+function getNumberOfParentFolders(path, root = "html") {
+	var parsed = path.split(root + '/');
+	var counter = 0;
+	if(parsed[1]) {
+		var parsed_even_more = parsed[1].split('/');
+		counter = parsed_even_more.length;
+	}
+	return counter;
+}
+/**
+ * 
+ * @param {Array} obj 
+ */
+function transpileInterviewNavigation(objects) {
+	str_builder = ``;
+	objects.map(obj => {
+		str_builder += `<li id="${obj.Name.replace(" ", "_")}_menu" class="nav-elem"><div>${obj.Name}</div></li>\n`;		
+	});
+	return str_builder;
+}
+
+function transpileJsonInterviewCardsToHTML(enJson, skJson) {
+	var file;
+	var dir = getDirname() + "/components/";
+	var langs = [];
+	var files = [];
+	var card_deck_start = `
+	<div class="card-deck interview">`;
```

Comment:
Actually, this has to be not because I like it but because I'm not indenting the javascript code but the HTML code that will be produced. For the HTML to be correctly indented it has to be this way. I'm sorry for that, if you have better idea around this I'm open for sharing.

Reply:
Can you figure out how to have both nicely formatted? :)